django/contrib/admin/options.py

```diff
@@ -2013,12 +2013,15 @@ def changelist_view(self, request, extra_context=None):
                 changecount = 0
                 for form in formset.forms:
                     if form.has_changed():
-                        obj = self.save_form(request, form, change=True)
-                        self.save_model(request, obj, form, change=True)
-                        self.save_related(request, form, formsets=[], change=True)
-                        change_msg = self.construct_change_message(request, form, None)
-                        self.log_change(request, obj, change_msg)
-                        changecount += 1
+                        with transaction.atomic(using=router.db_for_write(self.model)):
```

Comment:
The main question is about the expected behavior. Should we instead rollback all changes when an error appears? :thinking: (\cc @carltongibson @vzima) i.e.
```python
with transaction.atomic(using=router.db_for_write(self.model)):
    for form in formset.forms:
        ...
```

Reply:
According to me, as each form is independent from each other, I think we should not rollback all the changes.

Reply:
I think partial updates is tricky thing to get right. How do we indicate to the user which rows were successfully updated and which not? I think for a simple first-addition the top-level transaction is more appropriate

Reply:
> +1 for an all-or-nothing policy.

+1 from me too. @Shubh1815 Please update the patch.